web/app/js/components/TopRoutes.jsx

```diff
@@ -155,22 +167,26 @@ class TopRoutes extends React.Component {
     this.props[`onChange${_upperFirst(name)}`](formVal);
   }
 
-  handleNamespaceSelect = e => {
+  handleNamespaceSelect = nsKey => e => {
     let query = this.state.query;
     let formVal = _get(e, 'target.value');
-    query.namespace = formVal;
-    this.handleUrlUpdate("namespace", formVal);
+    query[nsKey] = formVal;
+    this.handleUrlUpdate(nsKey, formVal);
     this.setState({ query });
   };
 
-  handleResourceSelect = e => {
+  handleResourceSelect = (nameKey, typeKey) => e => {
     let query = this.state.query;
     let resource = _get(e, 'target.value');
-    let [resource_type, resource_name] = resource.split("/");
-    query.resource_name = resource_name;
-    query.resource_type = resource_type;
-    this.handleUrlUpdate("resource_name", resource_name);
-    this.handleUrlUpdate("resource_type", resource_type);
+    let [resourceType, resourceName] = resource.split("/");
+
+    resourceName = resourceName || "";
+    query[nameKey] = resourceName;
+    this.handleUrlUpdate(nameKey, resourceName);
+
+    query[typeKey] = resourceType;
+    this.handleUrlUpdate(typeKey, resourceType);
```

Comment:
nit, maybe for another PR, what about modifying `handleUrlUpdate` to take all of `state.query`, and then calling it once at the end of both `handleNamespaceSelect` and `handleResourceSelect`?